code/datums/datumvars.dm

```diff
@@ -715,7 +715,10 @@
 		log_admin("[key_name(usr)] has offered control of ([key_name(M)]) to ghosts.")
 		var/minhours = input(usr, "Minimum hours required to play [M]?", "Set Min Hrs", 10) as num
 		message_admins("[key_name_admin(usr)] has offered control of ([key_name_admin(M)]) to ghosts with [minhours] hrs playtime")
-		var/list/mob/dead/observer/candidates = pollCandidates("Do you want to play as [M.real_name]?", poll_time = 100, min_hours = minhours)
+		var/mname = M.name
+		if(M.real_name)
+			mname = M.real_name
+		var/list/mob/dead/observer/candidates = pollCandidates("Do you want to play as [mname]?", poll_time = 100, min_hours = minhours)

```

Comment:
Consider the ternary operator as an alternative that doesn't require an extra var and an awkward if in the middle of this block.

ie;
`var/list/mob/dead/observer/candidates = pollCandidates("Do you want to play as [M.real_name ? M.real_name : M.name]?", poll_time = 100, min_hours = minhours)`

Reply:
Alternatively, we could just initialize real_name to be whatever name is for all mobs, and then just overwrite that as needed.

Reply:
I vaguely recall something about the use of the ternary operator being frowned upon, possibly because it is less clear than this method (many people don't know what that operator means). I might be wrong on that, though. Not sure. If the maints want me to use it here, I will. In general, though, I don't use the ternary operator anywhere.

I'm more skeptical about initializing mob.real_name to equal mob.name during New() for all mobs. While I agree that it would be a good thing to do as a general practice, I suspect there are mobs that override New() without calling ..(), and as such that solution won't work with those mobs unless I find them and modify them individually. I might be wrong, but I felt like going that route would still leave open the possibility that you'd still get "do you want to play as ?" for some mobs. This solution, on the other hand, is certain to work for all mobs. That's why I did it this way.

Reply:
What @Dave-TH has is far cleaner and better.

Reply:
Implemented @Dave-TH 's suggestion.